src/coprocessor/dag/batch_executor/scan_executor.rs

```diff
@@ -0,0 +1,516 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![allow(unused)]
+
+use kvproto::coprocessor::KeyRange;
+use tipb::expression::FieldType;
+use tipb::schema::ColumnInfo;
+
+use crate::storage::{Key, Store};
+
+use super::interface::*;
+use crate::coprocessor::codec::batch::LazyBatchColumnVec;
+use crate::coprocessor::dag::expr::EvalContext;
+use crate::coprocessor::dag::Scanner;
+use crate::coprocessor::Result;
+
+/// A trait that determines how to deal with possible point ranges.
+pub trait PointRangePolicy: Send + Sync + 'static {
+    fn is_point_range(&self, range: &KeyRange) -> bool;
+}
+
+/// A policy that respects point ranges.
+///
+/// Notice that this exists because currently we use `prefix_next()` result as end_key to mark a
+/// range as point range. Once we have other clearer notation, e.g. a flag, we wont't need these
```

Comment:
```suggestion
/// range as point range. Once we have other more clear notation, e.g. a flag, we won't need these
```

Reply:
Looks like both are correct: https://english.stackexchange.com/questions/2166/more-clear-vs-clearer-when-to-use-more-instead-of-er

Reply:
`wont't` should still be fixed in any case.